### PR TITLE
Wk 236 validation

### PR DESCRIPTION
**Pull request by aditudorache**
This PR implements the field validation of the location form. The logic for show/hide components depending on values is implemented on the form level. A review en improvement interation will follow as soon as the time will allow that.

### CODE REVIEW

File: src/components/locationForm/components/FormInput.js [hunk 1]
```diff
@@ -1,22 +1,47 @@
 import React from 'react';
-import styled from '@datapunt/asc-core';
-import { Label, themeSpacing } from '@datapunt/asc-ui';
+import styled, { css } from '@datapunt/asc-core';
+import { Label, themeSpacing, themeColor } from '@datapunt/asc-ui';
 
 const LabelInput = styled(Label)`
+  ${({ hidden }) =>
```
Comment (janjaap) [inline src/components/locationForm/components/FormInput.js]: Or
```jsx
display: ${({ hidden }) => hidden ? 'none' : 'flex'};
```

File: src/components/locationForm/components/TextAreaInput.jsx [hunk 2]
```diff
@@ -16,42 +16,23 @@ const StyledInput = styled(Input)`
...
-    onChange(e);
-  };
-
+const TextAreaInput = ({ name, value, onChange, ...otherProps }) => {
   return (
     <StyledInput
       as="textarea"
       name={name}
-      value={value}
+      value={value || ''}
```
Comment (janjaap) [inline src/components/locationForm/components/TextAreaInput.jsx]: Why have an empty string here? The prop `value` is a required prop.

File: src/components/locationForm/components/TextInput.jsx [hunk 3]
```diff
@@ -16,35 +16,25 @@ const StyledInput = styled(Input)`
...
-    setValue(e.target.value);
-    onChange(e);
-  };
-
+const TextInput = ({ name, value, onChange, ...otherProps }) => {
   return (
     <StyledInput
       name={name}
-      value={value}
+      value={value || ''}
```
Comment (janjaap) [inline src/components/locationForm/components/TextInput.jsx]: The `value` prop already has a default value. Is this logic operator necessary?

File: src/components/locationForm/definitions/FormFields.js [hunk 4]
```diff
@@ -1,81 +1,126 @@
...
     name: 'coordinaten',
     label: 'Coördinaten',
     Component: TextInput,
     width: '80%',
+    validation: {
+      required: REQUIRED_MESSAGE,
+      pattern: {
+        value: /^\d{1,2}.\d{7}, \d{1,2}.\d{7}$/,
+        message:
+          'De coordinaten zijn niet in de correct format `xx.xxxxxxx, x.xxxxxxx,` ',
```
Comment (janjaap) [inline src/components/locationForm/definitions/FormFields.js]: Moet zijn
```
'De coördinaten zijn niet in het correct format `xx.xxxxxxx, xx.xxxxxxx` '
```

File: src/components/locationForm/definitions/FormFields.js [hunk 5]
```diff
@@ -1,81 +1,126 @@
...
     label: 'Jaar opgenomen in blackspotlijst',
-    defaultValue: String(new Date().getFullYear()),
+    initialValue: String(new Date().getFullYear()),
+    visible: true,
+    Component: TextInput,
+    width: '30%',
+    validation: {
+      pattern: {
+        value: /^(\d{4})?$/,
+        message: 'Het jaar is niet in de correct format `jjjj` ',
```
Comment (janjaap) [inline src/components/locationForm/definitions/FormFields.js]: Moet zijn
```
'Het jaar is niet in het correcte format `jjjj`'
```

File: src/components/locationForm/definitions/FormFields.js [hunk 6]
```diff
@@ -1,81 +1,126 @@
...
+      },
+    },
+  },
+  {
+    column: 1,
+    name: 'jaar_ongeval_quickscan',
+    label: 'Jaar opgenomen in de protocol',
```
Comment (janjaap) [inline src/components/locationForm/definitions/FormFields.js]: Moet zijn
```
'Jaar opgenomen in het protocol'
```

File: src/components/locationForm/definitions/FormFields.js [hunk 7]
```diff
@@ -1,81 +1,126 @@
...
+    initialValue: String(new Date().getFullYear()),
+    visible: false,
     Component: TextInput,
     width: '30%',
+    validation: {
+      pattern: {
+        value: /^(\d{4})?$/,
+        message: 'Het jaar is niet in de correct format `jjjj` ',
```
Comment (janjaap) [inline src/components/locationForm/definitions/FormFields.js]: Moet zijn
```
'Het jaar is niet in het correcte format `jjjj`'
```

File: src/components/map/Map.js [hunk 8]
```diff
@@ -1,6 +1,4 @@
 import React, { useState, useEffect } from 'react';
-import L from 'leaflet';
-import icon from 'leaflet/dist/images/marker-icon.png';
 
 // Imports needed for amaps
 import 'leaflet/dist/leaflet.css';
```
Comment (janjaap) [inline src/components/map/Map.js]: Would you still need the css if the you don't need leaflet?

File: src/components/locationForm/LocationForm.jsx [hunk 9]
```diff
@@ -116,26 +153,27 @@ const LocationForm = ({ id }) => {
...
+              ({ id, name, ...otherProps }) =>
+                visible[name] && (
+                  <FormInput
+                    key={id}
+                    name={name}
+                    onChange={handleChange}
+                    value={values[name]}
+                    error={errors[name]}
+                    {...otherProps}
+                  ></FormInput>
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/LocationForm.jsx]: 👍🏻

File: src/components/locationForm/LocationForm.jsx [hunk 10]
```diff
@@ -81,15 +114,17 @@ const LocationForm = ({ id }) => {
...
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/LocationForm.jsx]: 👍🏻

File: src/components/locationForm/LocationForm.jsx [hunk 11]
```diff
@@ -1,37 +1,73 @@
...
+    () =>
+      id
+        ? {
+            ...initalValues,
+            ...location,
+          }
+        : {
+            ...initalValues,
+          },
+    []
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/LocationForm.jsx]: of course, thanks!

File: src/components/locationForm/LocationForm.jsx [hunk 12]
```diff
@@ -99,15 +134,17 @@ const LocationForm = ({ id }) => {
...
+              ({ id, name, ...otherProps }) =>
+                visible[name] && (
+                  <FormInput
+                    key={id}
+                    name={name}
+                    onChange={handleChange}
+                    value={values[name]}
+                    error={errors[name]}
+                    {...otherProps}
+                  ></FormInput>
```
[the remark this answers is not in the document]
Comment (aditudorache) [reply src/components/locationForm/LocationForm.jsx]: 👍🏻

File: src/components/locationForm/components/FormInput.js (hunk 1 above)
Comment (aditudorache) [reply src/components/locationForm/components/FormInput.js]: nice!

File: src/components/locationForm/definitions/FormFields.js (hunk 4 above)
Comment (aditudorache) [reply src/components/locationForm/definitions/FormFields.js]: thanks!

File: src/components/locationForm/definitions/FormFields.js (hunk 6 above)
Comment (aditudorache) [reply src/components/locationForm/definitions/FormFields.js]: thanks

File: src/components/locationForm/definitions/FormFields.js (hunk 7 above)
Comment (aditudorache) [reply src/components/locationForm/definitions/FormFields.js]: thanks

File: src/components/map/Map.js (hunk 8 above)
Comment (aditudorache) [reply src/components/map/Map.js]: moved to the useMap hook.

File: src/components/locationForm/components/TextAreaInput.jsx (hunk 2 above)
Comment (aditudorache) [reply src/components/locationForm/components/TextAreaInput.jsx]: yes, this helps prevent assigning of the null values.

File: src/components/locationForm/components/TextInput.jsx (hunk 3 above)
Comment (aditudorache) [reply src/components/locationForm/components/TextInput.jsx]: yes, this helps prevent assigning of the null values.

**Pull request merged**